fontbe/src/mvar.rs

```diff
@@ -0,0 +1,329 @@
+            return Ok(());
+        }
+        let locations = sources.keys().cloned().collect::<BTreeSet<_>>();
+        let model = self.models.entry(locations).or_insert_with(|| {
+            // this glyph defines its own set of locations, a new sparse model is needed
+            VariationModel::new(sources.keys().cloned().collect(), self.axes.clone()).unwrap()
+        });
+        let deltas: Vec<_> = model
+            .deltas(&sources)
+            .map_err(|e| Error::MvarDeltaError(mvar_tag, e))?
+            .into_iter()
+            .filter_map(|(region, values)| {
+                if region.is_default() {
+                    return None;
+                }
+                // Only 1 value per region for our input
+                assert!(values.len() == 1, "{} values?!", values.len());
+                Some((
+                    region.to_write_fonts_variation_region(&self.axes),
+                    values[0].ot_round(),
+                ))
+            })
+            .collect();
+        // don't encode no-op deltas
+        if deltas.iter().all(|(_, delta)| *delta == 0) {
+            return Ok(());
+        }
+        self.deltas.insert(mvar_tag, deltas);
+        Ok(())
+    }
+
+    fn build(self) -> Option<Mvar> {
+        let mut builder = VariationStoreBuilder::new(self.axes.len() as u16);
+        let delta_ids = self
+            .deltas
+            .into_iter()
+            .map(|(tag, deltas)| (tag, builder.add_deltas(deltas)))
+            .collect::<Vec<_>>();
+
+        let (varstore, index_map) = builder.build();
+
+        let records = delta_ids
+            .into_iter()
+            .map(|(tag, temp_id)| {
+                let varidx = index_map.get(temp_id).unwrap();
+                ValueRecord::new(
+                    tag,
+                    varidx.delta_set_outer_index,
+                    varidx.delta_set_inner_index,
+                )
+            })
+            .collect::<Vec<_>>();
+
+        (!records.is_empty()).then(|| Mvar::new(MajorMinor::VERSION_1_0, Some(varstore), records))
+    }
+}
+
+impl Work<Context, AnyWorkId, Error> for MvarWork {
+    fn id(&self) -> AnyWorkId {
+        WorkId::Mvar.into()
+    }
+
+    fn read_access(&self) -> Access<AnyWorkId> {
+        AccessBuilder::new()
+            .variant(FeWorkId::StaticMetadata)
+            .variant(FeWorkId::GlobalMetrics)
+            .build()
+    }
+
+    /// Generate [MVAR](https://learn.microsoft.com/en-us/typography/opentype/spec/MVAR)
+    fn exec(&self, context: &Context) -> Result<(), Error> {
+        let static_metadata = context.ir.static_metadata.get();
+        let metrics = context.ir.global_metrics.get();
+        let var_model = &static_metadata.variation_model;
+
+        let mut mvar_builder = MvarBuilder::new(var_model.clone());
+        for (metric, values) in metrics.iter() {
+            // maybe we should get rid of GlobalMetric variants that aren't MVAR-relevant?
```

Comment:
Suggest making this an issue rather than a comment

Reply:
there are some global metrics that do not have an MVAR tag, notably hhea ascender/descender/lineGap (these are irrelevant and supposed to match the respective OS/2 metrics in a VF which implies USE_TYPO_METRICS); and the non-qualified (neither OS/2 nor hhea) ascender/descender, which both source formats have and is mostly used to encode the "real" height of the ascender/descender, used to define hinting zones (as opposed to the one that gets written in the font and used for line spacing purposes with all the baggage of legacy and weird recommendations like "linegap must be 0 otherwise BAD THINGS happen, etc."); we actually use the real ascender/descender to draw the default .notdef glyph, it's better to keep doing that than trust whatever gets into OS/2 or hhea.
So I decided that I'm not going to force GlobalMetrcis to map to what MVAR supports but allow the mvar_tag() to keep returning an Option<Tag>, and skip whatever metric is not associated with an MVAR tag. No need to file issues.